pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -585,6 +585,11 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
             realtimeRowsConsumedMeter =
                 _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1,
                     realtimeRowsConsumedMeter);
+            long currentTime = System.currentTimeMillis();
+            long pinotIngestionDelayMs = currentTime - msgMetadata.getRecordIngestionTimeMs();
```

Comment:
can you reconfirm the semantics of `getRecordIngestionTime()`? I am not sure if that is supposed to mean the last hop ingestion time or original ingestion time (remember, this part is stream agnostic, so it needs to hold good for all streams -- Kafka, Kinesis, Pulsar, pubsub,...)

Reply:
done this is the ingestion time, just what we need